src/VestRewardsDistributionJob.sol

```diff
@@ -0,0 +1,178 @@
+    event Work(bytes32 indexed network, address rewDist, uint distAmounts);
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event AddRewardDistribution(address indexed rewDist, uint256 delay);
+    event RemoveRewardDistribution(address indexed rewDist);
+    event ModifiedDistributionDelay(address indexed rewDist, uint256 delay);
+
+    constructor(
+        address _sequencer,
+        uint256 _minimumDelay
+    ) {
+        sequencer = SequencerLike(_sequencer);
+        minimumDelay = _minimumDelay;
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    // --- Reward Distribution Admin ---
+    function addRewardDistribution(address rewDist, uint256 delay) external auth {
+        if (!distributions.add(rewDist)) revert RewardDistributionExists(rewDist);
+        if (delay < minimumDelay) revert LessThanMinimumDelay(delay);
+        distributionDelays[rewDist] = delay;
+        emit AddRewardDistribution(rewDist, delay);
+    }
+
+    function removeRewardDistribution(address rewDist) external auth {
+        if (!distributions.remove(rewDist)) revert RewardDistributionDoesNotExist(rewDist);
+        delete distributionDelays[rewDist];
+        emit RemoveRewardDistribution(rewDist);
+    }
+
+    function modifyDistributionDelay(address rewDist, uint256 delay) external auth {
+        if (!distributions.contains(rewDist)) revert RewardDistributionDoesNotExist(rewDist);
+        if (delay < minimumDelay) revert LessThanMinimumDelay(delay);
+        distributionDelays[rewDist] = delay;
+        emit ModifiedDistributionDelay(rewDist, delay);
+    }
+
+    function work(bytes32 network, bytes calldata args) public {
+        if (!sequencer.isMaster(network)) revert NotMaster(network);
+
+        (address rewDist) = abi.decode(args, (address));
+
+        // prevent keeper from calling random contracts having distribute()
+        if (!distributions.contains(rewDist)) revert RewardDistributionDoesNotExist(rewDist);
+        // ensure that the right delay has elapsed
+        if (canDistributeAfter(rewDist) >= block.timestamp) revert CannotDistributeYet(rewDist);
+        // we omit checking the unpaid amount because if it is 0 it will revert during distribute()
+        uint256 distAmount = VestedRewardsDistributionLike(rewDist).distribute();
+        emit Work(network, rewDist, distAmount);
+    }
+
+    function workable(bytes32 network) external view override returns (bool, bytes memory) {
+        if (!sequencer.isMaster(network)) return (false, bytes("Network is not master"));
+
+        uint256 distributionsLen = distributions.length();
+        if (distributionsLen > 0) {
+            address distributable;
+            // this is used to find the distribute() than could have been called the earliest
+            // we use a hack that ensures the right functionality while avoiding an extra check for 0 value
+            uint256 earliestDistCall = block.timestamp + timeMagicNumber;
```

Comment:
I'm not sure if I'm following the logic here. What is your intent?
Do you want to give priority to the rewards distribution contract that is overdue `distribute()` the longest?
Maybe this is not relevant, because the checks will happen on every block. You can simply return the first one you find that matches the criteria and return early.